packages/transaction-controller/src/types.ts

```diff
@@ -1569,12 +1569,6 @@ export type TransactionBatchRequest = {
   /** Transactions to be submitted as part of the batch. */
   transactions: TransactionBatchSingleRequest[];
 
-  /**
-   * Whether to use the publish batch hook to submit the batch.
-   * Defaults to false.
-   */
-  useHook?: boolean;
```

Comment:
We shouldn't need `useHook` anymore but we still need some optional customisation here.

By default, we want to use 7702 if available, then the hook, then sequential.

But for gasless for example, we want to only use the smart transactions hook.

For EIP-5792, right now we only want 7702.

Perhaps the cleanest and simplest way to specify that is with three optional booleans, `disable7702`, `disableHook`, and `disableSequential` so the caller can be more specific if needed?

I considered a `methods` array with a new enum but that sounds more awkward for the caller, given array length, and duplicates etc.